### PR TITLE
FALCON-1795. Kill api not killing waiting/ready instances

### PR DESCRIPTION
**Pull request by sandeepSamudrala**

### CODE REVIEW

File: oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java [hunk 1]
```diff
@@ -1628,6 +1638,16 @@ private void resume(String cluster, String jobId) throws FalconException {
         }
     }
 
+    private void kill(String cluster, String jobId, String rangeType, String scope) throws FalconException {
+        try {
+            OozieClientFactory.get(cluster).kill(jobId, rangeType, scope);
```
Comment (pavankumar526) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: Did you miss to attach LocalOozieCoordClientProxy class ?

Comment (sandeepSamudrala) [inline oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java]: Yes. I have updated the PR. Thanks @pavankumar526  for pointing it out.

File: oozie/src/main/java/org/apache/oozie/client/ProxyOozieClient.java [hunk 2]
```diff
@@ -286,6 +286,25 @@ public Object call() throws Exception {
     }
 
     @Override
+    public List<CoordinatorAction> kill(final String jobId, final String rangeType, final String scope)
```
Comment (pavankumar526) [inline oozie/src/main/java/org/apache/oozie/client/ProxyOozieClient.java]: Not Required

File: unit/src/test/java/org/apache/falcon/unit/FalconUnitTestBase.java [hunk 3]
```diff
@@ -285,6 +287,29 @@ public void createData(String feedName, String cluster, String time,
...
+    public void deleteData(String feedName, String cluster) throws FalconException, ParseException, IOException {
+        Entity existingEntity = configStore.get(EntityType.FEED, feedName);
+        if (existingEntity == null) {
+            throw new FalconException("Feed Not Found  " + feedName);
+        }
+        Feed feed = (Feed) existingEntity;
+        Storage rawStorage = FeedHelper.createStorage(cluster, feed);
+        String feedPathTemplate = rawStorage.getUriTemplate(LocationType.DATA);
+
+        Properties vars = new Properties();
```
Comment (pavankumar526) [inline unit/src/test/java/org/apache/falcon/unit/FalconUnitTestBase.java]: Can't we directly delete feed base path ? Why getting all files and delete ?

File: oozie/src/main/java/org/apache/oozie/client/ProxyOozieClient.java (hunk 2 above)
Comment (sandeepSamudrala) [inline oozie/src/main/java/org/apache/oozie/client/ProxyOozieClient.java]: yes. Removed the method.

File: unit/src/test/java/org/apache/falcon/unit/FalconUnitTestBase.java (hunk 3 above)
Comment (sandeepSamudrala) [inline unit/src/test/java/org/apache/falcon/unit/FalconUnitTestBase.java]: Makes sense, made the respective changes .

**Comment by pavankumar526** [comment]
Looks good +1

**Comment by peeyushb** [comment]
+1. LGTM.

**Comment by ajayyadava** [comment]
:+1:

**Pull request closed without merging**